CRM/Core/Config/Runtime.php

```diff
@@ -32,7 +32,7 @@
  * the DSN, CMS type, CMS URL, etc. Generally, runtime properties must be
  * determined externally (before loading CiviCRM).
  */
-class CRM_Core_Config_Runtime {
+class CRM_Core_Config_Runtime extends CRM_Core_Config_MagicMerge {
```

Comment:
@monishdeb , some good/bad things about this patch
 * Good
    * I think it's right to remove the `set_include_path()` stuff from `CRM_Core_Config_Runtime::initialize()`. Generally, the behavior of `Runtime` is to read information about the runtime environment (read CIVICRM_UF, CIVICRM_DSN, etc). Reading generally shouldn't have any side-effects (like changing the `include_path` or making directories). To the extent there are side-effects, these are undesirable leftovers (from the refactoring process). It makes sense to move the `set_include_path` somewhere else.
 * Bad
    * Extending the class `MagicMerge` seems too heavy. `MagicMerge` is a pretty a thick/complex class, and it's primary purpose is providing backwards-compatibility for the hundreds of random things which rely on `CRM_Core_Config::singleton()`. (It looks like `$config` but actually combines `runtime`, `settings`, `services`, etc.) There should only one be one instance of `MagicMerge` within a request.
   * In this case, it seems like it's extending `MagicMerge` so that you can call `getSettings()`. The normal/simpler way to get settings is through `Civi::settings()`. No need to extend anything.
      * Note: I'm a little embarrassed about `MagicMerge::getSettings()` and `$this->settings`. It was a micro-optimization, added under the theory that `$settings` would be used *very frequently* within `MagicMerge`. (Like a few hundred times per request.) This is an odd thing to do and breaks from the convention of using `Civi::settings()`, which is why it's `protected`.